### PR TITLE
fix UnboundLocalError instead of proper error message

### PR DESCRIPTION
**Pull request by Czaki**
fix for
```
Traceback (most recent call last):
  File "/home/czaki/.pyenv/versions/3.6.8/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/czaki/.pyenv/versions/3.6.8/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/czaki/Projekty/cibuildwheel/cibuildwheel/__main__.py", line 254, in <module>
    main()
  File "/home/czaki/Projekty/cibuildwheel/cibuildwheel/__main__.py", line 76, in main
    if platform is None:
UnboundLocalError: local variable 'platform' referenced before assignment
```

### CODE REVIEW

File: cibuildwheel/__main__.py [hunk 1]
```diff
@@ -73,6 +72,8 @@ def main():
                 platform = 'macos'
             elif sys.platform == 'win32':
                 platform = 'windows'
+            else:
+                platform = None
```
Comment (mayeut) [inline cibuildwheel/__main__.py]: the assignment shall be moved to the same level as the `ci` assignment to fix all cases. This only fixes the case where we're running in CI but `sys.platform` returns an unsupported value.
If we wan't a different message for each case, could be modified as follow:
```
        ci = strtobool(os.environ.get('CI', 'false')) or 'BITRISE_BUILD_NUMBER' in os.environ or 'AZURE_HTTP_USER_AGENT' in os.environ
        if not ci:
            print('cibuildwheel: Unable to detect platform. cibuildwheel should run on your CI server, '
                  .........,
                  file=sys.stderr)
            exit(2)
        if sys.platform.startswith('linux'):
            platform = 'linux'
        elif sys.platform == 'darwin':
            platform = 'macos'
        elif sys.platform == 'win32':
            platform = 'windows'
        else:
            print('cibuildwheel: Unable to detect platform from "sys.platform"'
                  .........,
                  file=sys.stderr)
            exit(2)

```

Comment (Czaki) [reply cibuildwheel/__main__.py]: I change it, but in another way. Your proposition create code repetition.

Comment (mayeut) [reply cibuildwheel/__main__.py]: @Czaki, in fact you're using my exact first proposition. The second proposition doesn't duplicate code.
Full picture:
```

    detect_obsolete_options()

    if args.platform != 'auto':
        platform = args.platform
    else:
        ci = strtobool(os.environ.get('CI', 'false')) or 'BITRISE_BUILD_NUMBER' in os.environ or 'AZURE_HTTP_USER_AGENT' in os.environ
        if not ci:
            print('cibuildwheel: Unable to detect platform. cibuildwheel should run on your CI server, '
                  'Travis CI, AppVeyor, Azure Pipelines and CircleCI are supported. You can run on your '
                  'development machine or other CI providers using the --platform argument. Check --help '
                  'output for more information.',
                  file=sys.stderr)
            exit(2)
        if sys.platform.startswith('linux'):
            platform = 'linux'
        elif sys.platform == 'darwin':
            platform = 'macos'
        elif sys.platform == 'win32':
            platform = 'windows'
        else:
            print('cibuildwheel: Unable to detect platform from "sys.platform" in a CI environment. You can run '
                  'cibuildwheel using the --platform argument. Check --help output for more information.',
                  file=sys.stderr)
            exit(2)

    output_dir = args.output_dir
```

**Comment by YannickJadoul** [comment]
Oh, whoops, regression introduced in #204?

**Comment by mayeut** [comment]
> Oh, whoops, regression introduced in #204?

Yes, as mentioned in [this comment](https://github.com/joerick/cibuildwheel/pull/214/files#r347129283)

**Comment by Czaki** [comment]
I'm writing tests for this now.

File: cibuildwheel/__main__.py (hunk 1 above)
Comment (Czaki) [reply cibuildwheel/__main__.py]: Ok, I see the difference and agree that it is better.

**Comment by mayeut** [comment]
@Czaki,

Those tests are almost exactly what I had in mind. Can you move them in a new file please (maybe `unit_test/platform_test.py` or `unit_test/parse_platform_test.py`) ? the `unit_test/environment_test.py` file is used for `cibuildwheel.environment` unit tests.

File: cibuildwheel/__main__.py [hunk 2]
```diff
@@ -61,25 +61,29 @@ def main():
     args = parser.parse_args()
 
     detect_obsolete_options()
-
```
Comment (mayeut) [inline cibuildwheel/__main__.py]: Maybe this empty should be kept for readability ? allows the whole logic block for platform detection to be on its own.

Comment (Czaki) [reply cibuildwheel/__main__.py]: Ok.  maybe this function should be renamed to `detect_and_fix_obsolete_environment_variables` or `fix_deprecated_environment_variables"?

Comment (mayeut) [reply cibuildwheel/__main__.py]: I like the second one `fix_deprecated_environment_variables`,  it's clear enough. No need to add `detect_and_`.

Comment (Czaki) [reply cibuildwheel/__main__.py]: this PR or another? it is not connected with main problem solved by this PR.

Comment (mayeut) [reply cibuildwheel/__main__.py]: Let's say another. Since @joerick approved, I'll merge this one.

**Comment by Czaki** [comment]
@mayeut  A moment ago one question show in my head. If the `build` functions from `linux` `macos` and `windows` should be monkeypatched? For security?

**Comment by mayeut** [comment]
@Czaki, I'm not sure to understand what you mean. You can Open an Issue if you have a concern, we'll exchange on the subject in that issue.

**Pull request merged**